### PR TITLE
Skilltrees 2022 demo

### LINKED ISSUE #1462: Create demo file
We will create a demo file, using the example from last year to better understand what must be done, which reflects our work from this year. This will entail looking at last year's demo, examining what work we can use from this year in the demo, and creating the new file using this information.

team members: Nikola, Arya, Annika

### PR DESCRIPTION
**Pull request by nbilaver**
This is the PR for the demo. It goes over Complex Skills, the work on class prefabs and other class contributions made by skilltrees, and readers. The PR consists of the demo file and a .md file acting as the script.

### CODE REVIEW

File: src/skilltrees/examples/2022-Demo-Script.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+#Skilltrees Demo 2022 Script
+This file is to give the order of the functions needed to run the Skilltrees Demo, and build instructions. 
+
+Lore:
+You are a young adventurer, going to skills academy in order to get strong. 
+Your mentor teaches you about complex skills, classes, and how to read, when he turns out to be vampire, and you kill him.
```
Comment (chelseaegbarin) [inline src/skilltrees/examples/2022-Demo-Script.md]: to be a vampire*

File: src/skilltrees/examples/skilltrees_demo_2022.c [hunk 2]
```diff
@@ -0,0 +1,677 @@
...
+    "     |      ╚════██║██╔═██╗ ██║██║     ██║     ██║   ██╔══██╗██╔══╝  ██╔══╝  ╚════██║       |\n"
+    "     |      ███████║██║  ██╗██║███████╗███████╗██║   ██║  ██║███████╗███████╗███████║       |\n"
+    "     |      ╚══════╝╚═╝  ╚═╝╚═╝╚══════╝╚══════╝╚═╝   ╚═╝  ╚═╝╚══════╝╚══════╝╚══════╝       |\n"
+    "     |   ___________________________________________________________________________________|___\n"
+    "     |  /                                                                                      /\n"
+    "     \\_/______________________________________________________________________________________/\n";
+/* ========================================================================== */
+/* Declare skill tree ======================================================= */
+/* ========================================================================== */
+skill_tree_t* skill_treedemo;
```
Comment (chelseaegbarin) [inline src/skilltrees/examples/skilltrees_demo_2022.c]: I would keep snake case consistent and call this skill_tree_demo

File: src/skilltrees/examples/skilltrees_demo_2022.c [hunk 3]
```diff
@@ -0,0 +1,677 @@
...
+                            "Room progression is always (GO) NORTHward.");
+    chiventure_ctx_t *ctx = chiventure_ctx_new(game);
+
+    /* Create example rooms */
+    room_t* start_room = room_new("Start Room", "", "Once you get past this door, you will meet "
+                                                    "your new mentor!");
+
+    room_t* combined_room = room_new("Combined Skill Room", "", "You see your new mentor, in the middle of a large hallway.");
+
+    room_t* sequential_room = room_new("Sequential Skill Room", "", "You enter a room with a training dummy.\nYour mentor begins to train "
```
Comment (chelseaegbarin) [inline src/skilltrees/examples/skilltrees_demo_2022.c]: teach you* as opposed to train you

**Review by chelseaegbarin — APPROVED** [review]
Looks great, I loved the storyline!! Very creative and fun. I only had minor suggestions.

File: src/skilltrees/examples/skilltrees_demo_2022.c (hunk 3 above)
Comment (nbilaver) [reply src/skilltrees/examples/skilltrees_demo_2022.c]: Fixed!

File: src/skilltrees/examples/skilltrees_demo_2022.c (hunk 2 above)
Comment (nbilaver) [reply src/skilltrees/examples/skilltrees_demo_2022.c]: Fixed!

File: src/skilltrees/examples/2022-Demo-Script.md (hunk 1 above)
Comment (nbilaver) [reply src/skilltrees/examples/2022-Demo-Script.md]: Fixed!

File: src/skilltrees/examples/skilltrees_demo_2022.c [hunk 4]
```diff
@@ -0,0 +1,677 @@
...
+    print_to_cli(ctx, "Mentor: Over the last year, we have made even more powerful, "
+                        "complicated skills!");
+    print_to_cli(ctx, "Mentor: In this room, you can learn Combined "
+                        "Skills! Many smaller skills wrapped into one.");
+    print_to_cli(ctx, "Mentor: Lets teach you a large stat boost!");
+    return "";
+}
+
+//Random Skills lecture -- random code wasn't merged soon enough to use in the 
+//demo, however we put a brief acknowledgement of it here
```
Comment (jacknugent1529) [inline src/skilltrees/examples/skilltrees_demo_2022.c:564]: If you'd like to include random skills, you can update this now

**Review by jacknugent1529 — COMMENTED** [review]
This is a fantastic demo! If you'd like to include random skills in it, you could update it now. If you don't have the time to do that, that's fine. Just let me know and then I'll merge it

File: src/skilltrees/examples/skilltrees_demo_2022.c (hunk 4 above)
Comment (anchristensen) [reply src/skilltrees/examples/skilltrees_demo_2022.c:564]: I think we're a bit too busy to add in a random room, so we're going to keep it as is. Feel free to merge it when you have the chance!

**Review by jacknugent1529 — APPROVED** [review]
Looks good to me!

**Comment by jacknugent1529** [comment]
Issue Score: Excellent

Comments: Excellent work! The demo looks great.

**Pull request merged**